### ISSUE TITLE
Actions on unknown Message Type

### ISSUE THREAD

**Issue by martinduke** [moq-wg/moq-transport#241]
Fail, or pass over?

**Comment by kixelated** [comment]
IMO fail. We have version negotiation and should add extension negotiation so it should never happen.

The problem with ignoring messages is that MoqTransport is designed for relays. If an intermediate relay doesn't support a message, it then gets silently dropped.

**Comment by martinduke** [comment]
If it's going to Fail, then most Message types don't need a length field at all (OBJECT and ANNOUNCE_OK would be the exceptions)

**Comment by afrind** [comment]
Individual Comment:

Is there any reasonable action a relay can take on an unknown message, besides discarding it?  See how HTTP recently added a "Capsule" concept which is an envelop with a default relay action of "forward", which allows for end-to-end extensions without upgrading all intermediaries.

Given that MoQ handles forwarding on a per-track level, I think there probably isn't any way to generically forward a message that isn't associated with an active subscription.

That said, it makes me wonder if we want to add a capsule-like extension message, or does the existing OBJECT definition give enough flexibility?

**Comment by kixelated** [comment]
> Given that MoQ handles forwarding on a per-track level, I think there probably isn't any way to generically forward a message that isn't associated with an active subscription.

Yeah exactly. It's impossible to know the intended destination of an unknown message with the current ANNOUNCE/SUBSCRIBE mechanisms. 

I'm using the `PATH` (as the track namespace) and `ROLE` instead of `ANNOUNCE`. This actually does let you forward unknown messages from publisher -> subscribers and subscribers -> publisher since there's a 1:1 relationship with sessions and track namespaces. Things might be a bit weird with multiple publishers however.

**Comment by fluffy** [comment]
I think we need extensibility. We want the protocol to be extensible so a given server might support extension A and C and another server supports extension A and B.  It is very hard to do that using just a version number.

**Comment by ianswett** [comment]
There was support for not requiring a length and failing on unknown message types at the Interim, so the intent is to land PR #255 after review.
